src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -1379,3 +1529,229 @@ void CustomTextLayout::_OrderRuns()
 }
 
 #pragma endregion
+
+// Routine Description:
+// - Calculates the box drawing scale/translate matrix values to fit a box glyph into the cell as perfectly as possible.
+// Arguments:
+// - format -
+// - face -
+// - effect -
+// - fontScale - (optional, 1.0f default), if the given font face is going to be scaled versus the format, we need to know so we can compensate for that.
+// Return Value:
+// - S_OK, GSL/WIL errors, DirectWrite errors, or math errors.
+[[nodiscard]] HRESULT STDMETHODCALLTYPE CustomTextLayout::s_CalculateBoxEffect(IDWriteTextFormat* format, IDWriteFontFace1* face, IBoxDrawingEffect** effect, float fontScale) noexcept
+{
+    // The format is based around the main font that was specified by the user.
+    // We need to know its size as well as the final spacing that was calculated around
+    // it when it was first selected to get an idea of how large the bounding box is.
+    const auto fontSize = format->GetFontSize();
+
+    DWRITE_LINE_SPACING_METHOD spacingMethod;
+    float lineSpacing; // total height of the cells
+    float baseline; // vertical position counted down from the top where the characters "sit"
+    RETURN_IF_FAILED(format->GetLineSpacing(&spacingMethod, &lineSpacing, &baseline));
+
+    const float ascentPixels = baseline;
+    const float descentPixels = lineSpacing - baseline;

```

Comment:
Hey, uh, pal. We're never checking the exact pixel count of the advance. Maybe we need to feed that in here to ensure that the horizontal adjustment below relative to the cell goes correctly?

Reply:
Done.